internal/log/logger.go

```diff
@@ -0,0 +1,62 @@
+package log
+
+import (
+	"fmt"
+	"os"
+)
+
+var (
+	DebugLogFn LogFunc = nil
+	SQLLogFn   LogFunc = nil
+	ErrorLogFn LogFunc = nil
+)
+
+type LogFunc = func(string, ...interface{})
```

Comment:
Why is the argument signature `(string, ...interface{})` if the intent is to be compatible with `log.Println` (https://golang.org/pkg/log/#Println)?

Reply:
Its actually not for log.Println, because log.Println is not a structured logger. Over the last few years, the go community has mostly settled on the notion that structured loggers (not printf ones) should adhere to either `func(...interface{})` a la go-kit, or `func(string, ...interface{})` a la zap, log15, logxi. I went with the latter because it allows performance benefits like msg level de-duping, and makes the old logxi calls a direct drop in, no adapter func needed, along with zap.Sugared which is the most common structured logging library I've seen in the wild.